mm2src/coins/lp_coins.rs

```diff
@@ -1201,8 +1236,33 @@ pub async fn lp_coininit(ctx: &MmArc, ticker: &str, req: &Json) -> Result<MmCoin
             )
             .into()
         },
+        CoinProtocol::BCH { slp_prefix } => {
+            let prefix = try_s!(CashAddrPrefix::from_str(&slp_prefix));
+            let bch = try_s!(bch_coin_from_conf_and_request(ctx, ticker, &coins_en, req, prefix, secret).await);
+            bch.into()
+        },
+        CoinProtocol::SLPTOKEN {
+            platform,
+            token_id,
+            decimals,
+            required_confirmations,
+        } => {
+            let platform_coin = try_s!(lp_coinfind(ctx, &platform).await);
+            let platform_coin = match platform_coin {
+                Some(MmCoinEnum::Bch(coin)) => coin,
+                Some(_) => return ERR!("Platform coin {} is not BCH", platform),
+                None => return ERR!("Platform coin {} is not activated", platform),
+            };
+
+            let confs = required_confirmations.unwrap_or(platform_coin.required_confirmations());
+            let token = SlpToken::new(*decimals, ticker.into(), token_id.clone().into(), platform_coin, confs);
+            token.into()
+        },
         #[cfg(all(not(target_arch = "wasm32"), feature = "zhtlc"))]
-        CoinProtocol::ZHTLC => try_s!(z_coin_from_conf_and_request(ctx, ticker, &coins_en, req, secret).await).into(),
+        CoinProtocol::ZHTLC => {
+            let dbdir = ctx.dbdir();
+            try_s!(z_coin_from_conf_and_request(ctx, ticker, &coins_en, req, secret, dbdir).await).into()
```

Comment:
I think here the `?` can be used if the trait is implemented.

Reply:
We still have some old code depending on `try_s!` - I will try to get rid of it wherever possible, thanks!